src/session.c

```diff
@@ -107,20 +137,23 @@ static int
 banner_receive(LIBSSH2_SESSION * session)
 {
     ssize_t ret;
+    int rc;
     size_t banner_len;
 
     if(session->banner_TxRx_state == libssh2_NB_state_idle) {
         banner_len = 0;
 
+        if(session->remote.banner)
+            LIBSSH2_FREE(session, session->remote.banner);
+        session->remote.banner = NULL;
+
         session->banner_TxRx_state = libssh2_NB_state_created;
     }
     else {
         banner_len = session->banner_TxRx_total_send;
     }
 
-    while((banner_len < sizeof(session->banner_TxRx_banner)) &&
-          ((banner_len == 0)
-           || (session->banner_TxRx_banner[banner_len - 1] != '\n'))) {
+    while(1) {
```

Comment:
```suggestion
    for(;;) {
```
seem preferred in src.